learn/models/bootstrap-metrics/index.qmd

```diff
@@ -0,0 +1,212 @@
+
+tidymodels_prefer()
+theme_set(theme_bw())
+options(pillar.advice = FALSE, pillar.min_title_chars = Inf)
+registerDoMC(cores = parallel::detectCores(logical = FALSE))
+```
+
+We'll use the [delivery time data](https://modeldata.tidymodels.org/reference/deliveries.html) and follow the analysis used on [this website](https://aml4td.org/chapters/whole-game.html#sec-delivery-times). The outcome is the time for food to be delivered, and the predictors include the day/hour of the order, the distance, and what was included in the order (columns starting with `item_`):
+
+```{r}
+library(tidymodels)
+
+str(deliveries)
+```
+
+Given the amount of data, a validation set was used _in lieu_ of multiple resamples. This means that we can fit models on the training set, evaluate/compare them with the validation set, and reserve the test set for a final performance assessment (after model development). The data splitting code is:
+
+```{r}
+#| label: splitting
+
+set.seed(991)
+delivery_split <- initial_validation_split(deliveries, prop = c(0.6, 0.2), 
+                                           strata = time_to_delivery)
+
+# Make data frames for each of the three data sets
+delivery_train <- training(delivery_split)
+delivery_test  <- testing(delivery_split)
+delivery_val   <- validation(delivery_split)
+
+# Create an object that makes a validation set "look like resampling"
+# to tidymodels: 
+delivery_rs    <- validation_set(delivery_split)
+```
+
+## Tuning a Model
+
+To demonstrate, we'll use a multivariate adaptive regression spline ([MARS](https://scholar.google.com/scholar?hl=en&as_sdt=0%2C7&q=%22multivariate+adaptive+regression+splines%22&btnG=)) model produced by the earth package. The original analysis of these data shows some significant interactions between predictors, so we will specify a model that can estimate them using the argument `prod_degree = 2`. Let's create a model specification that tunes the number of terms to retain: 
+ 
+```{r}
+#| label: mars-spec
+
+mars_spec <- 
+  mars(num_terms = tune(), prod_degree = 2, prune_method = "none") %>% 
+  set_mode("regression")
+```
+
+Let's use grid search to evaluate a grid of values between 2 and 50. We'll use `tune_grid()` with an option to save the out-of-sample (i.e., validation set) predictions for each candidate model in the grid. By default, for regression models, the function computes the root mean squared error (RMSE) and R<sup>2</sup>:  
+
+```{r}
+#| label: grid
+#| cache: true
+
+grid <- tibble(num_terms = 2:50)
+
+ctrl <- control_grid(save_pred = TRUE)
+
+mars_res <- 
+  mars_spec %>% 
+  tune_grid(
+    time_to_delivery ~ .,
+    resamples = delivery_rs,
+    grid = grid,
+    control = ctrl
+  )
+```
+
+How did the model look?
+
+```{r}
+#| label: perf-plot
+autoplot(mars_res)
+```
+
+After about 20 retained terms, both statistics appear to plateau.  However, we have no sense of the noise around these values. Is a model with 20 terms just as good as a model using 50? 
```

Comment:
Having read further, I see that you were very intentional about the higher number of terms. Maybe put the motivation for which model you're interested in up here instead of where it is now. Still wondering why you'd want more terms though :)